app/controllers/posts_controller.rb

```diff
@@ -20,4 +20,10 @@ def show
     @post = Post.find(request.parameters[:id])
     @user = User.find(params[:user_id])
   end
+
+  def destroy
+    @post = Post.find(request.parameters[:id])
+    @post.destroy
+    redirect_to all_post_path(params[:user_id])
+  end
 end
```

Comment:
- [ ] Please make sure to use `CanCanCan` for Post authorization in Post Controller. You can consider the below code.

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```